Can't compile

Can't compile. Wrong getcwd usage.

Output:
$ make
mkdir bin -p
gcc -Wall -Werror -lm -Iinclude src/main.c -c
gcc -Wall -Werror -lm -Iinclude src/reading.c -c
gcc -Wall -Werror -lm -Iinclude src/ui.c -c
src/ui.c: In function ‘user’:
src/ui.c:25:17: error: argument 1 is null but the corresponding size argument 2 value is 256 [-Werror=nonnull]
   25 |     char *pwd = getcwd(NULL, 256);
      |                 ^~~~~~~~~~~~~~~~~
In file included from src/ui.c:3:
/usr/include/unistd.h:520:14: note: in a call to function ‘getcwd’ declared with attribute ‘write_only (1, 2)’
  520 | extern char *getcwd (char *__buf, size_t __size) __THROW __wur
      |              ^~~~~~
cc1: all warnings being treated as errors
make: *** [Makefile:18: ui] Error 1

--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 #переменная CC указывает на компилятор
 CC=gcc
 #переменная CFLAGS хранит необходимые для компиляции флаги
-CFLAGS=-Wall -Werror -lm -Iinclude
+CFLAGS=-Wall -lm -Iinclude
 
 .PHONY: clean, all
 
@@ -37,5 +37,5 @@ OBJ_RM: main.o reading.o ui.o execute.o pidList.o
 clean:
 	rm bin/shell
 	#rm bin/reading
-	rmdir /bin
+	rmdir bin
 	#ls | grep -v "\." | grep -v Makefile | xargs rm

--- a/src/ui.c
+++ b/src/ui.c
@@ -22,7 +22,7 @@ char *getCurrentDir(char *pwd) {
 }
 
 void user() {
-    char *pwd = getcwd(NULL, 256);
+    char *pwd = getcwd(NULL, 0);
     char *userName = getenv("USER");
     char host[256];
     gethostname(host, _SC_HOST_NAME_MAX);

